components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/username/NotificationUsernameRecoveryManager.java

```diff
@@ -1,211 +1,125 @@
+import org.wso2.carbon.identity.recovery.IdentityRecoveryException;
+import org.wso2.carbon.identity.recovery.IdentityRecoveryServerException;
+import org.wso2.carbon.identity.recovery.internal.IdentityRecoveryServiceDataHolder;
+import org.wso2.carbon.identity.recovery.mapping.UsernameConfig;
+import org.wso2.carbon.identity.recovery.util.Utils;
+
+import java.util.List;
+
+/**
+ * Manager class which can be used to recover passwords using a notification.
+ */
+public class NotificationUsernameRecoveryManager {
+
+    private static final Logger log = LoggerFactory.getLogger(NotificationUsernameRecoveryManager.class);
+    private static NotificationUsernameRecoveryManager instance = new NotificationUsernameRecoveryManager();
+
+    private static UsernameConfig usernameConfig;
+
+    private NotificationUsernameRecoveryManager() {
+    }
+
+    public static NotificationUsernameRecoveryManager getInstance() {
+        try {
+            usernameConfig = Utils.getRecoveryConfigs().getRecovery().getUsername();
+        } catch (IdentityRecoveryServerException e) {
+            log.error("Error while Loading recovery-config file.", e);
+        }
+        return instance;
+    }
+
+    public boolean verifyUsername(List<Claim> claims) throws
+            IdentityRecoveryException {
+
+        return recoverUserByClaims(claims);
+
+    }
+
+
+    // TODO trigger Notification Method
+
+    public boolean recoverUserByClaims(List<Claim> claims)
+            throws IdentityRecoveryException {
+
+        /* No need of checking recovery enable from back end side it is already checked from API side
+         And portal side.
+
+         */
+
+        if (claims == null || claims.size() < 1) {
+
+            if (log.isDebugEnabled()) {
+                log.debug("No claims are recieved");
+            }
+            return false;
+            //TODO send exception
+        }
+
+        User user;
+        List<User> resultedUserList = getUserList(claims);
+
+
+        if (resultedUserList.size() == 1) {
+            user = resultedUserList.get(0);
+            if (log.isDebugEnabled()) {
+                log.debug("There are more than one user in the result set : "
+                        + user.toString());
+            }
+
+            //TODO send the username to the user as an email
+            return true;
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("There are more than one user in the result set : "
+                        + resultedUserList.toString());
+            }
+            return false;
+
+        }
+
+
+    }
+
+    private static List<User> getUserList(List<Claim> claims) throws IdentityRecoveryException {
+
+        RealmService realmService = IdentityRecoveryServiceDataHolder.getInstance().getRealmService();
+        IdentityStore identityStore = realmService.getIdentityStore();
+
+
+        try {
+            return identityStore.listUsers(claims, 0, 3);
```

Comment:
It's possible to miss users if the claims are distributed in multiple connectors.